### PR TITLE
Allow dediated-admin to manage project request templates

### PR DESCRIPTION
**Pull request by fahlmant**
From the ticket, this PR does the following:

-Remove SSS that creates / manages template "osd-project-request"
-Remove SSS that patches projects.config.openshift.io "cluster" resource
-Add RBAC to allow dedicated-admins to manage project.config.openshift.io

### CODE REVIEW

File: deploy/rbac-permissions-operator-config/05-dedicated-admins-aggregate-cluster.ClusterRole.yaml [hunk 1]
```diff
@@ -270,3 +270,13 @@ rules:
   verbs:
   - use
 ### END
+### Start - manage project.config.openshift.io CR
+# https://issues.redhat.com/browse/OSD-5566
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - "project"
```
Comment (jewzaam) [inline deploy/rbac-permissions-operator-config/05-dedicated-admins-aggregate-cluster.ClusterRole.yaml:278]:
```suggestion
  - projects
```

**Comment by fahlmant** [comment]
https://issues.redhat.com/browse/OSD-5566

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: deploy/osd-project-request-template/04-patch.label-default-namespace.yaml [hunk 2]
```diff
@@ -1,7 +0,0 @@
-kind: Namespace
-apiVersion: v1
-name: default
-applyMode: AlwaysApply
-patch: |-
-  { "metadata": {"labels": {"name":"default"} } }
```
Comment (jewzaam) [inline deploy/osd-project-request-template/04-patch.label-default-namespace.yaml:6]: Can't take this out without knowing who is using it.  Probably added to support NetworkPolicy for an internal customer.

File: deploy/osd-project-request-template/02-role.dedicated-admins-project-request.yaml [hunk 3]
```diff
@@ -1,12 +0,0 @@
-apiVersion: rbac.authorization.k8s.io/v1
-kind: Role
-metadata:
-  name: dedicated-admins-project-request
```
Comment (jewzaam) [inline deploy/osd-project-request-template/02-role.dedicated-admins-project-request.yaml:4]: Can't take this out, it removes management of the template, which we're trying to allow.

File: deploy/osd-project-request-template/03-rolebinding.dedicated-admins-project-request.yaml [hunk 4]
```diff
@@ -1,15 +0,0 @@
-apiVersion: rbac.authorization.k8s.io/v1
-kind: RoleBinding
-metadata:
-  name: dedicated-admins-project-request
```
Comment (jewzaam) [inline deploy/osd-project-request-template/03-rolebinding.dedicated-admins-project-request.yaml:4]: Need this to realize management of the template for the customer.

File: deploy/rbac-permissions-operator-config/05-dedicated-admins-aggregate-cluster.ClusterRole.yaml [hunk 5]
```diff
@@ -270,3 +270,12 @@ rules:
...
   - use
 ### END
+### Start - manage project.config.openshift.io CR
+# https://issues.redhat.com/browse/OSD-5566
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - projects
+  verbs:
+  - "*"
```
Comment (jewzaam) [inline deploy/rbac-permissions-operator-config/05-dedicated-admins-aggregate-cluster.ClusterRole.yaml:280]:
```suggestion
  - get
  - list
  - patch
  - update
  - watch
```
Probably better to not allow delete, and therefore not allow create.

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Comment by fahlmant** [comment]
Closing in favor of #528

**Pull request closed without merging**